dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+	            }
+
+	            // to check user has permission to publish this page
+	            boolean permission = APILocator.getPermissionAPI().doesUserHavePermission(htmlPage, PERMISSION_PUBLISH, backendUser);
+	            context.put("permission", new Boolean(permission));
+
+	            // Check if the user is a CMS Administrator
+	            boolean adminUser = false;
+	            try {
+
+
+	                String adminRoleKey = "";
+	                try {
+	                    Role adminRole = APILocator.getRoleAPI().loadRoleByKey(Config.getStringProperty("CMS_ADMINISTRATOR_ROLE", "CMS Administrator"));
+	                    adminRoleKey = adminRole.getRoleKey();
+	                } catch (Exception e) {
+	                }
+
+	                Role[] userRoles = (Role[]) APILocator.getRoleAPI().loadRolesForUser(backendUser.getUserId()).toArray(new Role[0]);
+	                for (int i = 0; i < userRoles.length; i++) {
+	                    Role userRole = (Role) userRoles[i];
+	                    if (userRole.getRoleKey().equals(adminRoleKey)) {
+	                        adminUser = true;
+	                    }
+	                }
+	            } catch (Exception e) {
+	            }
+	            context.put("cmsAdminUser", new Boolean(adminUser));
+
+	        }
+
+	        // gets pageChannel for this path
+	        String pageChannel = UtilMethods.getPageChannel(idURI);
+	        context.put("pageChannel", pageChannel);
+	        context.put("PREVIEW_MODE", new Boolean(mode == PageMode.PREVIEW_MODE));
+	        context.put("EDIT_MODE", new Boolean(mode == PageMode.EDIT_MODE));
+	        context.put("ADMIN_MODE", new Boolean(mode.isAdmin));
+
+	        // for publish button on admin control
+	        // I HAVE TO FIX THIS!!!!! FOR THE NEW ONE
+	        context.put("TEMPLATE_LIVE_CONTENT", new Boolean(false));
+	        context.put("CONTAINER_LIVE_CONTENT", new Boolean(false));
+	        context.put("CONTENTLET_LIVE_CONTENT", new Boolean(false));
+
+
+	
+
+	        context.put("livePage", mode.showLive);
+
+	        context.put("language", (String) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE));
+
+	        if (mode.isAdmin) {
+
+	            // Making sure you are viewing the latest list of languages on any
+	            // admin mode
+	            languages = APILocator.getLanguageAPI().getLanguages();
+	            context.put("languages", languages);
+
+	            // gets user id from request for mod user
+	            com.liferay.portal.model.User backendUser = null;
+
+	            try {
+	                backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
+	                // Skin skin = backendUser.getSkin();
+	                // context.put("USER_SKIN", skin.getSkinId());
+	                context.put("backendUser", backendUser);
+	            } catch (Exception nsue) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)